### PR TITLE
Update ansible provisioner triggers

### PR DESCRIPTION
**Pull request by SweetOps**
## What
* Update ansible provisioner triggers

## Why
* Need fix misstate in previous version

### CODE REVIEW

**Comment by SweetOps** [comment]
@ osterman, yes, you're right `external data sources`   works with JSON. I'll rework script for returning   the JSON. This's just initial PR.

File: main.tf [hunk 1]
```diff
@@ -1,8 +1,17 @@
+data "external" "md5sum" {
+  program = ["sh", "${path.module}/md5sum.sh"]
```
Comment (osterman) [inline main.tf]: Per the TF documentation: https://www.terraform.io/docs/providers/external/data_source.html

> This mechanism is provided as an "escape hatch" for exceptional situations where a first-class Terraform provider is not more appropriate.

Terraform natively supports compression. We use it other modules. Use this instead if a shell script.

Comment (osterman) [reply main.tf]: https://www.terraform.io/docs/providers/archive/d/archive_file.html

File: md5sum.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+set -ueo pipefail
+query_path=$(cat | jq -r '.path')
+md5=$(tar -cf - $query_path | md5sum| cut -d' ' -f1)
```
Comment (osterman) [inline md5sum.sh]: This is not portable. E.g. on OSX it's just `md5` and not `md5sum`. Therefore, use `archive_file` resource instead.

File: main.tf [hunk 3]
```diff
@@ -1,12 +1,23 @@
+data "archive_file" "md5sum" {
```
Comment (osterman) [inline main.tf]: this should just be `default`

File: main.tf [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
+  depends_on = ["data.archive_file.md5sum"]
 
   triggers {
-    signature = "${sha256(dirname(var.playbook))}"
+    signature = "${data.archive_file.md5sum.output_md5}"
     command   = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
   }
 
   provisioner "local-exec" {
+    command = "rm -rf ${path.module}/playbook.zip"
```
Comment (osterman) [inline main.tf]: this should refer to `archive_file .default.output_path`

File: main.tf [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
+  depends_on = ["data.archive_file.default"]
 
   triggers {
-    signature = "${sha256(dirname(var.playbook))}"
+    signature = "${data.archive_file.default.output_md5}"
     command   = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
   }
 
   provisioner "local-exec" {
+    command = "rm -rf ${data.archive_file.default.output_path}"
```
Comment (osterman) [inline main.tf]: A file will never be recursive (`-r`)

File: main.tf [hunk 6]
```diff
@@ -1,12 +1,23 @@
+data "archive_file" "default" {
+  type        = "zip"
+  source_dir  = "${dirname(var.playbook)}"
+  output_path = "${path.module}/playbook.zip"
```
Comment (comeanother) [inline main.tf]: Add this file to `.gitignore`.

**Pull request merged**